In [1]:
import pandas as pd
import numpy as np
import EDA

## Cargar información

In [2]:
data_path_367= 'Datos/Estacion_meteorologica_367_2025-12-01_2026-04-30.csv'#Meteorológica vaisala
data_path_803= 'Datos/Estacion_nivel_803_2025-12-01_2026-04-30.csv'#Nivel
data_path_6004= 'Datos/Estacion_piranometro_6004_2025-12-01_2026-04-30.csv'#Piranómetro
data_path_35= 'Datos/Estacion_pluviometrica_35_2025-12-01_2026-04-30.csv'#Pluviómetro

In [3]:
def lectura_csv(path):
    data=pd.read_csv(path, sep=',',header=0)
    return data

In [4]:
data_367=lectura_csv(data_path_367)
data_803=lectura_csv(data_path_803)
data_6004=lectura_csv(data_path_6004)
data_35=lectura_csv(data_path_35)

## Descripción general de los datos

In [ ]:
data_367.describe()

,codigo,h,t,pr,vv,vv_max,dv,dv_max,p,calidad
count,217397.0,217397.000000,217397.000000,217397.000000,217397.000000,217397.000000,217397.000000,217397.000000,217397.000000,217397.000000
mean,367.0,81.449923,19.330236,815.977350,0.729525,1.310210,178.254383,187.781920,0.001236,5.121018
std,0.0,14.095989,3.034755,1.920536,0.541494,0.936224,106.000683,111.564761,0.010912,78.323360
min,367.0,35.940000,13.450000,809.510000,0.000000,0.000000,0.000000,0.000000,0.000000,1.000000
25%,367.0,71.000000,17.000000,814.800000,0.370000,0.700000,79.000000,73.000000,0.000000,1.000000
50%,367.0,86.000000,18.300000,816.200000,0.600000,1.000000,189.000000,216.000000,0.000000,1.000000
75%,367.0,93.000000,21.400000,817.300000,0.940000,1.600000,275.000000,279.000000,0.000000,1.000000
max,367.0,100.000000,28.600000,821.800000,13.470000,14.300000,359.000000,360.000000,0.520000,1567.000000


In [15]:
data_803.describe()

,codigo,nivel,calidad
count,217318.0,217318.000000,217318.000000
mean,803.0,8.727745,133.102039
std,0.0,36.802595,423.548420
min,803.0,-7.099976,1.000000
25%,803.0,1.500000,1.000000
50%,803.0,2.700012,1.000000
75%,803.0,4.299988,1.000000
max,803.0,344.299988,2502.000000


In [16]:
data_6004.describe()

,codigo,radiacion,calidad
count,214824.0,214824.00000,214824.000000
mean,6004.0,211.70116,13.742827
std,0.0,330.20899,141.384847
min,6004.0,0.00000,1.000000
25%,6004.0,0.00000,1.000000
50%,6004.0,0.00000,1.000000
75%,6004.0,335.80000,1.000000
max,6004.0,1611.03000,1583.000000


In [17]:
data_35.describe()

,codigo,p1,p2,calidad
count,217440.0,217440.000000,217440.000000,217440.000000
mean,35.0,-0.028260,-0.028213,1.706305
std,0.0,5.668344,5.668349,32.367241
min,35.0,-999.000000,-999.000000,1.000000
25%,35.0,0.000000,0.000000,1.000000
50%,35.0,0.000000,0.000000,1.000000
75%,35.0,0.000000,0.000000,1.000000
max,35.0,2.286000,2.286000,2512.000000


Todas las estaciones presentan indicadores de calidad muestran indicadores de calidad diferentes a 1 u 2 (los cuales corresponden a datos confiables -Según lo hallado en los manuales de la red-) lo que implica que los datos entregados muestran datos dudosos sobre los cuales se debe de tener especial cuidado en el procesamiento estadístico de la estación para realizar los cálculos en función de los dats confiables únicamente, sin tener en cuenta los definidos como dudosos por parte del criterio experto del proyecto SIATA. 

Adicionalmente, de la descripción anterior se evidencia que solo la red pluviométrica muestra un valor sentinela correspondiente a -999, lo cual es de considerar en las funciones realizadas para la detección de los mismos. 

## Tarea A: Análisis exploratorio de datos (EDA)

A1. Perfilamiento: tipos, rangos, nulos, valores centinela, duplicados, orden y regularidad del
índice temporal.

In [6]:
EDA.perfilamiento_eda(
data_367,
nombre="Meteorológica 367",
columna_fecha="fecha_hora",
frecuencia="1min",
valores_centinela=None
)


PERFILAMIENTO EDA: Meteorológica 367

[1] INFORMACIÓN GENERAL
--------------------------------------------------------------------------------
Número de registros       : 217,397
Número de columnas        : 11
Memoria utilizada         : 18.24 MB
Filas duplicadas          : 0
% filas duplicadas        : 0.00%
Nulos explícitos totales  : 43

[2] TIPOS DE DATOS
--------------------------------------------------------------------------------
   columna      tipo_dato
    codigo          int64
fecha_hora datetime64[ns]
         h        float64
         t        float64
        pr        float64
        vv        float64
    vv_max        float64
        dv        float64
    dv_max        float64
         p        float64
   calidad          int64

[3] NULOS EXPLÍCITOS
--------------------------------------------------------------------------------
   columna  n_nulos porcentaje_nulos
    codigo       43            0.02%
fecha_hora       43            0.02%
         h       43           

In [7]:
EDA.perfilamiento_eda(
data_803,
nombre="Nivel 803",
columna_fecha="fecha_hora",
frecuencia="1min",
valores_centinela=None
)


PERFILAMIENTO EDA: Nivel 803

[1] INFORMACIÓN GENERAL
--------------------------------------------------------------------------------
Número de registros       : 217,318
Número de columnas        : 4
Memoria utilizada         : 6.63 MB
Filas duplicadas          : 0
% filas duplicadas        : 0.00%
Nulos explícitos totales  : 122

[2] TIPOS DE DATOS
--------------------------------------------------------------------------------
   columna      tipo_dato
    codigo          int64
fecha_hora datetime64[ns]
     nivel        float64
   calidad          int64

[3] NULOS EXPLÍCITOS
--------------------------------------------------------------------------------
   columna  n_nulos porcentaje_nulos
    codigo      122            0.06%
fecha_hora      122            0.06%
     nivel      122            0.06%
   calidad      122            0.06%

[4] VALORES CENTINELA
--------------------------------------------------------------------------------
No se encontraron valores centinela.

[5] R

In [8]:
EDA.perfilamiento_eda(
data_6004,
nombre="Piranómetro 6004",
columna_fecha="fecha_hora",
frecuencia="1min",
valores_centinela=None
)


PERFILAMIENTO EDA: Piranómetro 6004

[1] INFORMACIÓN GENERAL
--------------------------------------------------------------------------------
Número de registros       : 214,824
Número de columnas        : 4
Memoria utilizada         : 6.56 MB
Filas duplicadas          : 0
% filas duplicadas        : 0.00%
Nulos explícitos totales  : 2,631

[2] TIPOS DE DATOS
--------------------------------------------------------------------------------
   columna      tipo_dato
    codigo          int64
fecha_hora datetime64[ns]
 radiacion        float64
   calidad          int64

[3] NULOS EXPLÍCITOS
--------------------------------------------------------------------------------
   columna  n_nulos porcentaje_nulos
    codigo     2631            1.21%
fecha_hora     2631            1.21%
 radiacion     2631            1.21%
   calidad     2631            1.21%

[4] VALORES CENTINELA
--------------------------------------------------------------------------------
No se encontraron valores centinel

In [9]:
EDA.perfilamiento_eda(
data_35,
nombre="Pluviómetro 35",
columna_fecha="fecha_hora",
frecuencia="1min",
valores_centinela=None
)


PERFILAMIENTO EDA: Pluviómetro 35

[1] INFORMACIÓN GENERAL
--------------------------------------------------------------------------------
Número de registros       : 217,440
Número de columnas        : 5
Memoria utilizada         : 8.29 MB
Filas duplicadas          : 0
% filas duplicadas        : 0.00%
Nulos explícitos totales  : 0

[2] TIPOS DE DATOS
--------------------------------------------------------------------------------
   columna      tipo_dato
    codigo          int64
fecha_hora datetime64[ns]
        p1        float64
        p2        float64
   calidad          int64

[3] NULOS EXPLÍCITOS
--------------------------------------------------------------------------------
   columna  n_nulos porcentaje_nulos
    codigo        0            0.00%
fecha_hora        0            0.00%
        p1        0            0.00%
        p2        0            0.00%
   calidad        0            0.00%

[4] VALORES CENTINELA
----------------------------------------------------------